98.md

```diff
@@ -0,0 +1,68 @@
+NIP-98
+======
+
+HTTP Auth
+-------------------------
+
+`draft` `optional` `author:kieran` `author:melvincarvalho`
+
+This NIP defines an ephemerial event used to authorize requests to HTTP servers using nostr events.
+
+This is useful for HTTP services which are build for Nostr and deal with Nostr user accounts.
+
+## Nostr event
+
+A `kind 27235` (In reference to [RFC 7235](https://www.rfc-editor.org/rfc/rfc7235)) event is used.
+
+The `content` SHOULD be empty.
+
+The following tags are defined as REQUIRED.
+
+* `u` - absolute URL
+* `method` - HTTP Request Method
+
+Example event:
+```json
+{
+    "id": "fe964e758903360f28d8424d092da8494ed207cba823110be3a57dfe4b578734",
+    "pubkey": "63fe6318dc58583cfe16810f86dd09e18bfd76aabc24a0081ce2856f330504ed",
+    "content": "",
+    "kind": 27235,
+    "created_at": 1682327852,
+    "tags": [
+        [
+            "u",
+            "https://api.snort.social/api/v1/n5sp/list"
+        ],
+        [
+            "method",
+            "GET"
+        ]
+    ],
+    "sig": "5ed9d8ec958bc854f997bdc24ac337d005af372324747efe4a00e24f4c30437ff4dd8308684bed467d9d6be3e5a517bb43b1732cc7d33949a3aaf86705c22184"
+}
+```
+
+Servers MUST perform the following checks in order to validate the event:
+1. The `kind` MUST be `27235`.
+2. The `created_at` MUST be within a reasonable time window (suggestion 60 seconds).
+3. The `url` tag MUST be exactly the same as the absolute request URL (including query parameters).
```

Comment:
Yea i prefer `url` @melvincarvalho prefers `u`

Reply:
@melvincarvalho is wrong. 😃 That's my only remaining feedback, otherwise LGTM.

Reply:
Thanks for pointing that out. Indeed, we've previously discussed this and arrived at a consensus to use "u", rather than, "url"/"uri". Revisiting this decision would require effort since its already implemented, without providing substantial benefits.

Reply:
NIP-94 uses the "url" tag already. Having "u" and "url" coexist isn't ideal. https://github.com/nostr-protocol/nips/blob/master/94.md

Reply:
The feature has been already integrated as "u" in NosDAV, I'm using this alot in a lot of place, and it's going to stick. @v0l said he could live with this choice, leading to a consensus. A change at this stage would result in two conflicting NIP-98 type specs, which wouldn't be optimal. 

The statement, "The url tag MUST be exactly the same as the absolute request URL (including query parameters)", simply needs a correction from 'url' to 'u'.

Reply:
I am using it on two endpoints, but I would have no problem changing it. I'll adapt to what you all decide 🙂

Reply:
I have anchored this version to NAV-03 to ensure stability even if future changes occur, and have stated that NAV-03 is "based on NIP-98".

[NAV-03](https://github.com/nosdav/navs/blob/gh-pages/03.md)

There are compelling reasons for using "u" instead of "url", which will become clearer as NosDAV progresses towards the stable 0.1 version. Sincerely hope that this work will not necessitate a breaking change at this point.

Reply:
I support this NIP. 👍